Fixes to dynamical power spectrum + Dynamical Cross spectrum

matteobachetti opened the pull request:
+ Improves performance and consistency when using events (uses the internals already tested in `AveragePowerspectrum` to create light curves dynamically for each segment, rather than allocating the full light curve on the spot)
+ Resolves a bug when rebinning in time (it relied on dt, which was calculated in a weird manner)
+ Makes tests a little more understandable, I hope

BONUS: added Dynamical Cross spectrum, because why not

mgullik commented on stingray/crossspectrum.py:
```diff
@@ -1880,6 +1882,236 @@ def time_lag(self):
+            Time axis with new rebinned time resolution.
+
+        dynspec_new: numpy.ndarray
+            New rebinned Dynamical Power Spectrum.
```
New rebinned Dynamical Cross-Spectrum

mgullik commented on stingray/powerspectrum.py:
```diff
@@ -885,7 +883,7 @@ def initial_checks(self, *args, **kwargs):
 
 
-class DynamicalPowerspectrum(AveragedPowerspectrum):
+class DynamicalPowerspectrum(DynamicalCrossspectrum):
```
Also for this class I would specify the dt as a possible parameter if an EventList is passed as in the DynamicalCrossspectrum

mgullik commented on stingray/crossspectrum.py:
```diff
@@ -1880,6 +1882,236 @@ def time_lag(self):
+
+    dt: float
+        The time resolution.
+    """
```
Is it worth adding the possible methods in the docstring? 
I mean rebin_frequency and rebin_time

matteobachetti commented:
@mgullik yes, there is a confusion between two different meanings of `dt` (+`segment_size`), but `rebin_time` does what you say, not a rebinning of the light curve. I should have addressed your comments now, including renaming the input `dt` -> `sample_time`, which clears up at least part of the confusion.

mgullik commented on stingray/crossspectrum.py:
```diff
@@ -1880,6 +1882,253 @@ def time_lag(self):
+        """
+        Rebin the Dynamic Power Spectrum to a new time resolution.
+
+        Note: this is *not* the time resolution of the input light
```
"Note: this is *not* changing the time resolution of the input light"

matteobachetti replied:
Done

mgullik reviewed (approved):
Hi @matteobachetti, 

the review is completed, I suggested one last change to the docstring of the rebin_time. Sorry, probably too picky there 

One thing, since you were so thorough in implementing the DynamicalCrossspectrum I'm wondering why don't you implement an option to output a complex matrix with real and imaginary parts instead of the normalized squared absolute values of Fourier amplitudes? In this way, it's even possible to create a dynamic lag frequency spectrum (boom!) However, not something to hold this PR longer. 

Another very small thing about the tests, I noticed is that you always run the tests on DynamicalCrossspectrum with the same lightcurves for data1 and data2. Does it make sense to test it with two different light curves?

matteobachetti commented:
As far as I can see, I'm not using the squared amplitudes anywhere, and indeed, the Dynamical Cross spectrum has complex values when you feed it with different light curves in the two channels. I added a test to show explicitly that
